src/Version10/PluginInterface.php

```diff
@@ -4,12 +4,27 @@
 
 namespace Phpcq\PluginApi\Version10;
 
+use Phpcq\PluginApi\Version10\Configuration\ConfigurationOptionsBuilderInterface;
+
 /**
  * This interface describes the root for generic phpcq plugins.
  *
- * A generic plugin does nothing by itself and you therefore have to implement one or more of it's child interfaces.
+ * A generic plugin has a name and may describe it's configuration. A generic plugin does not specify which purpose it
+ * has by itself and you therefore have to implement one or more of it's child interfaces.
  */
 interface PluginInterface
 {
+    /**
+     * The unique name of the plugin.
+     *
+     * @return string
+     */
     public function getName(): string;
+
+    /**
+     * Describe available config options.
+     *
+     * @param ConfigurationOptionsBuilderInterface $configOptionsBuilder The config options builder.
+     */
+    public function describeOptions(ConfigurationOptionsBuilderInterface $configOptionsBuilder): void;
```

Comment:
Should maybe become an own interface `ConfigurablePlugin`? Otherwise non-configurable plugins won't be distinguished from configurable ones.